### ISSUE TITLE
Browser page title != Map Title... Currently

### ISSUE THREAD

**Issue by Bortseb** [metamaps/metamaps#645]
I just noticed that some pages still have titles like "Map 2230 | Metamaps" instead of the map name.

**Comment by Bortseb** [comment]
Ok @Connoropolous , point me in the right direction on this... better yet, think out loud so I can start getting a sense for how the code is organized on a higher level... and how you think through finding where you need to start reading in the code. (When you are considering working on a new issue)

**Comment by Connoropolous** [comment]
Ok, here's a few quick thoughts. 

When does this behaviour happen? When navigating from an Explore Maps page, to a Map Page... 
something to notice is that the page doesn't refresh when the link is clicked, even though it IS a link. That means that Metamaps.Router is coming into play, which handles these transitions between pages, when a refresh isn't happening. 
https://github.com/metamaps/metamaps/blob/develop/app/assets/javascripts/src/Metamaps.Router.js

There is one function in there that is handling all the logic of the transition from Explore page to Map page. By the time you identify that, you will be close to the heart of this issue...

At some point, it needs to fetch the map you are loading from the server, which doesn't happen in that file, but see if you can find your way to it. it will be once you have the map object that you can set the document title to the name of the map

**Comment by devvmh** [comment]
I can expand on where the title gets set. There are a few different paths.

If you load a map directly, Ruby on Rails picks the title. That code is in `app/views/layouts/_head.html.erb`, where it does `yield(:title)`. Then `app/views/maps/show.html.erb` sets it with this code: `content_for :title, @map.name + " | Metamaps"`.

But if you load any other page, and then navigate to the map, the title is not changed by Rails. Rather, when you click links the [Backbone Router](http://backbonejs.org/#Router) intercepts the navigation and instead does two things:

1) changes the url in the browser to the new url - we've configured this to match our rails routes closely
2) run some function to tear down the old page and load the new page

I guess I can talk through my thought process in trying to find where to solve this issue:

The current code is at [line 151 in the router file](https://github.com/metamaps/metamaps/blob/develop/app/assets/javascripts/src/Metamaps.Router.js#L151), within the `maps` function: `document.title = 'Map ' + id + ' | Metamaps'`

That function ends with `Metamaps.Map.launch(id)`, so we'll have to either move the document.title line or add a new one within Metamaps.Map.launch.

Metamaps.Map.launch's layout is a bit confusing; most of its definition is taken up by a new `start` function. That function is then called by this bit of code at the end:

```
    $.ajax({
      url: '/maps/' + id + '/contains.json',
      success: start
    })
```

So essentially, when you click the map, it calls Metamaps.Router.maps, which tears down the old page and gets the new map id, then it calls Metamaps.Map.launch(id), which does this ajax call, and then calls the `start` function. I think `start` is where you want to drop your code.

**Comment by Bortseb** [comment]
This issue can be closed if PR #655 is sufficient...
